### PR TITLE
[SPARK-30630][ML][2.4] Deprecate numTrees in GBT in 2.4.5

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Deprecate numTrees in GBT in 2.4.5 so it can be removed in 3.0.0

### Why are the changes needed?
Currently, GBT has
```
  /**
   * Number of trees in ensemble
   */
  @Since("2.0.0")
  val getNumTrees: Int = trees.length
```
and
```
  /** Number of trees in ensemble */
  val numTrees: Int = trees.length
```
I think we should remove one of them. I will deprecate it in 2.4.5 and remove it in 3.0.0

### Does this PR introduce any user-facing change?
Deprecate numTrees in 2.4.5

### How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @dongjoon-hyun @srowen

File: mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala [hunk 1]
```diff
@@ -319,7 +319,12 @@ class GBTClassificationModel private[ml](
     }
   }
 
-  /** Number of trees in ensemble */
+  /**
+   * Number of trees in ensemble
+   *
+   * @deprecated  Use [[getNumTrees]] instead. This method will be removed in 3.0.0
```
Comment (dongjoon-hyun) [inline mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala]: nit. `3.0.0` -> `3.0.0.`

File: mllib/src/main/scala/org/apache/spark/ml/regression/GBTRegressor.scala [hunk 2]
```diff
@@ -255,10 +255,15 @@ class GBTRegressionModel private[ml](
...
     val treePredictions = _trees.map(_.rootNode.predictImpl(features).prediction)
-    blas.ddot(numTrees, treePredictions, 1, _treeWeights, 1)
+    blas.ddot(getNumTrees, treePredictions, 1, _treeWeights, 1)
   }
 
-  /** Number of trees in ensemble */
+  /**
+   * Number of trees in ensemble
+   *
+   * @deprecated  Use [[getNumTrees]] instead. This method will be removed in 3.0.0
```
Comment (dongjoon-hyun) [inline mllib/src/main/scala/org/apache/spark/ml/regression/GBTRegressor.scala]: ditto.

**Comment by huaxingao** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @huaxingao .
Merged to branch-2.4.

**Comment by huaxingao** [comment]
Thank you very much!! @dongjoon-hyun

**Pull request closed without merging**